pkg/app/server/grpcapi/api.go

```diff
@@ -314,21 +314,15 @@ func (a *API) RenameApplicationConfigFile(ctx context.Context, req *apiservice.R
 	}
 
 	for _, appID := range req.ApplicationIds {
-		var denied bool
-		err := a.applicationStore.UpdateApplication(ctx, appID, func(app *model.Application) error {
-			if app.ProjectId != key.ProjectId {
-				denied = true
-				return fmt.Errorf("Requested application %s does not belong to your project", appID)
-			}
-			app.GitPath.ConfigFilename = req.NewFilename
-			return nil
-		})
+		app, err := a.applicationStore.Get(ctx, appID)
 		if err != nil {
-			if denied {
-				return nil, status.Error(codes.PermissionDenied, err.Error())
-			}
-			a.logger.Error("failed to update application", zap.Error(err))
-			return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to update application %s", appID))
+			return nil, gRPCEntityOperationError(err, fmt.Sprintf("failed to get application %s", appID))
+		}
+		if app.ProjectId != key.ProjectId {
+			return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("requested application %s does not belong to your project", appID))
```

Comment:
[imo] I think it better that  `PermissionDenied` is also handled by gRPCEntityOperationError. but it's up to you and in another PR.

Reply:
Oh, Let me fix it in another PR! so this PR lgtm!

Reply:
Yeh, the current gRPCEntityOperationError does not handle any errors other than datastore's error
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/grpcapi.go#L195-L208
So please address what you want by separated PR 🙏